cpp/generate_new_journal_alert_stats.cc

```diff
@@ -152,38 +154,35 @@ void CollectUsageStats(const std::string &user_type, Stats * const stats) {
 }
 
 
-void GenerateAndMailReport(const std::string &email_address, const Stats &stats) {
-    if (EmailSender::SimplerSendEmail("no-reply@ub.uni-tuebingen.de", email_address, "Journal Alert Stats",
-                                      "Host: " + DnsUtil::GetHostname() + "\n"
-                                      + "Report interval in days: " + std::to_string(stats.report_interval_in_days_) + "\n"
-                                      + "Number of users w/ subscriptions: "
-                                      + std::to_string(stats.no_of_users_with_subscriptions_) + "\n"
-                                      + "Average number of subscriptions per user: "
-                                      + StringUtil::ToString(stats.average_subscriptions_per_user_, 3) + "\n"
-                                      + "Average number of bundle subscriptions per user: "
-                                      + StringUtil::ToString(stats.average_number_of_bundle_subscriptions_, 3) + "\n"
-                                      + "Total number of currently subscribed journals: "
-                                      + std::to_string(stats.no_of_subscribed_journals_) + "\n"
-                                      + "Number of subscribed journals w/ notifications: "
-                                      + std::to_string(stats.no_of_subscribed_journals_with_notifications_) + "\n"
-                                      + "Average number of notified articles per notified journal: "
-                                      + StringUtil::ToString(stats.average_number_of_notified_articles_per_notified_journal_))
-        > 299)
-        LOG_ERROR("failed to send an email report to \"" + email_address + "\"!");
+void GenerateReport(File * const report_file, const Stats &stats) {
+    (*report_file) << "\"Host\"," << TextUtil::CSVEscape(DnsUtil::GetHostname()) << '\n'
+                   << "\"Report interval in days\"," << stats.report_interval_in_days_ << '\n'
```

Comment:
Missing ending enclosure `"` before each newline

Reply:
No, the numbers should not intended to be quoted.

Reply:
Wrong.

Reply:
Right now, the field has an opening quote without an ending quote. So you should either add the ending quote or remove the opening quote, but it can not stay as it is right now.